src/toxcore/friend_connection/packet/friend_requests.rs

```diff
@@ -0,0 +1,69 @@
+/*! Friend requests struct
+*/
+
+use nom::rest;
+
+use toxcore::binary_io::*;
+use toxcore::onion::packet::*;
+use toxcore::crypto_core::*;
+use toxcore::toxid::NoSpam;
+
+const ONION_SEND_BASE: usize =  PUBLICKEYBYTES + SIZE_IPPORT + MACBYTES;
+const ONION_SEND_1: usize = secretbox::NONCEBYTES + ONION_SEND_BASE * 3;
+const MAX_ONION_DATA_SIZE: usize = ONION_MAX_PACKET_SIZE - (ONION_SEND_1 + 1); // 1 is for packet_id
+const MIN_ONION_DATA_REQUEST_SIZE: usize = 1 + PUBLICKEYBYTES + secretbox::NONCEBYTES + PUBLICKEYBYTES + MACBYTES; // 1 is for packet_id
+/// Maximum size in butes of Onion Data Request packet
+const MAX_DATA_REQUEST_SIZE: usize = MAX_ONION_DATA_SIZE - MIN_ONION_DATA_REQUEST_SIZE;
+/// Minimum size in bytes of Onion Data Response packet
+const MIN_ONION_DATA_RESPONSE_SIZE: usize = PUBLICKEYBYTES + MACBYTES;
```

Comment:
Are you sure that this value is correct? It seems nonce is missing here.

Reply:
done